tests/e2e/api-interface.spec.ts

```diff
@@ -0,0 +1,59 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import { runCommand } from '../utils/command-exec';
+import { tempNamespace, k8s_deployment_name } from '../../tests/resources/test-constants';
+import { IUpgradeMessage } from '../../src/lib/upgrade-message';
+
+chai.use(chaiHttp);
+
+describe('The API', () => {
+
+  before(async () => {
+    process.env.CHT_NAMESPACE = tempNamespace;
+    process.env.CHT_DEPLOYMENT_NAME = k8s_deployment_name;
+    await runCommand(
+      `kubectl -n ${tempNamespace} apply -f tests/resources/alpine.yaml`,
+      'Creating an alpine deployment');
+    await runCommand(`sleep 10`, 'Waiting a few seconds...');
+  });
+
+  after(() => {
+    process.env.CHT_NAMESPACE = '';
+    process.env.CHT_DEPLOYMENT_NAME = '';
+  });
+
+  it('Listens on 5008', () => {
+    chai.request('http://localhost:5008')
+      .get('/')
+      .then((res) => {
+        expect(res).to.have.status(200);
+      });
+  });
+
+  it('Server status endpoint exists', () => {
+    chai.request('http://localhost:5008')
+      .get('/server-status')
+      .then((res) => {
+        expect(res).to.have.status(200);
+      });
+  });
+
+  it('Upgrade endpoint exists', () => {
+    chai.request('http://localhost:5008')
+      .get('/upgrade')
+      .then((res) => {
+        expect(res).to.not.have.status(404);
+      });
+  });
+
+  it('Should upgrade deployment', async () => {
+    const upgradeMessageArray: IUpgradeMessage[] = [{ containerName: 'alpine', imageTag: '3.15.0' }];
+
+    chai.request('http://localhost:5008')
+      .post('/upgrade')
+      .send(upgradeMessageArray)
+      .then((res) => {
+        expect(res).to.have.status(200);
```

Comment:
Please also assert on the status of the pods to verify that the containers have actually been updated.